README.md

```diff
@@ -2,3 +2,50 @@
 
 This repository contains the Radio Frequency Front End (RFFE) EPICS Input/Output
 Controller (IOC) for LNLS Beam Position Monitor (BPM).
+
+## Building the Docker image
+
+In order to build the Docker image, you shall execute the following command
+
+```bash
+docker build \
+  -t ghcr.io/lnls-dig/rffe-epics-ioc \
+  -f docker/Dockerfile \
+  --build-arg JOBS=$(nproc) \
+  $(pwd)
+```
+
+The `JOBS` argument specifies the number of jobs (1 by default) to use `make`
+the application (EPICS base, its modules and the IOC itself). It will not
+change the resulting image, only the time for building the image.
+
+## Running the IOC
+
+Afterwards, you should create a container specifying the crate and BPM slot
+numbers, RFFEs' base IP address, and set up the network to inherent the host's
+configuration. For instance,
+
+```bash
+docker run \
+  -d \
+  --network host \
+  --mount type=bind,source=/opt/autosave,target=/opt/autosave \
+  -e CRATE_NUMBER=23 \
+  -e BPM_NUMBER=1 \
+  -e RFFE_BASE_IP_ADDRESS=127.0.0.0 \
+  ghcr.io/lnls-dig/rffe-epics-ioc
+```
+
+`CRATE_NUMBER` and `BPM_NUMBER` environment vars are used to determine which
+area and device prefixes defined in the
+[iocBoot/iocBPMRFFEApp/slot-mapping.env](./iocBoot/iocBPMRFFEApp/slot-mapping.env)
+are to be used when launching the IOC. In addition, `BPM_NUMBER` is used to
+shift the `RFFE_BASE_IP_ADDRESS` in the host portion of the address,
+considering a `24` bits subnet mask (last dot-decimal), so that this matches
+the current network configuration.
```

Comment:
> These are implementation details and should not be documented here. I think this either belongs in a comment in the launch script or in a commit message (which is my preferred method).

Makes sense. I decided to include it along with the command to make it transparent that it is simplifying how to automatically compute IP addresses. We may remove it as this limitation will be fixed.

> The BPM_NUMBER explanation is a bit confusing. "Shift" to me reads as bit shifting, not just a number being added. I'm also slightly confused about why the netmask matters.

You're right. Poor choice of words there. About the netmask, I was assuming all the devices running in the same subnet. In this case, using a 24 bits netmask, we can only to change last dotted-decimal part of the address (which is currently supported by the implementation). But thinking about that now, this assumption does not need to be true for assigning addresses.

> One of our crates has tcp://192.168.2.201:9001 as the IP address for BPM 11. That works, because we can set the base as 190, but I think it might be helpful to support a more flexible addressing scheme in case we need to change things.

Any suggestions on that?

We could convert the address to binary, add the BPM offset, and convert it back to dotted-decimal. Something like `inet_ntoa(inet_aton(address) + bpm_number)`.

Reply:
> Makes sense. I decided to include it along with the command to make it transparent that it is simplifying how to automatically compute IP addresses. We may remove it as this limitation will be fixed.

I also meant how the crate and slot numbers are turned into PV prefixes.

> Any suggestions on that?

Not currently, no. That was more of a note, I think we can go ahead for now.